File: evidence/generated-systems/django/inventory-on-django/inventoryOnDjango/models/Warehouse.py
from django.db import models

#======================================================================
# 
# Encapsulates data for model Warehouse
#
# @author Harbormaster Dev Team
#
#======================================================================

#======================================================================
# Class Warehouse Declaration
#======================================================================
class Warehouse (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	name = models.CharField(max_length=200, null=True)
	code = models.CharField(max_length=200, null=True)
	address = Address
	timeZone = models.CharField(max_length=200, null=True)
	allowsOverAllocation = models.BooleanField(null=True)
	storageLocations = models.ManyToManyField('StorageLocation',  blank=True, related_name='+')
	inventoryItems = models.ManyToManyField('InventoryItem',  blank=True, related_name='+')
	inboundShipments = models.ManyToManyField('InboundShipment',  blank=True, related_name='+')
	outboundAllocations = models.ManyToManyField('OutboundAllocation',  blank=True, related_name='+')
	originTransfers = models.ManyToManyField('TransferOrder',  blank=True, related_name='+')
	destinationTransfers = models.ManyToManyField('TransferOrder',  blank=True, related_name='+')
	cycleCounts = models.ManyToManyField('CycleCount',  blank=True, related_name='+')

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.name
		str = str + self.code
		str = str + self.address
		str = str + self.timeZone
		str = str + self.allowsOverAllocation
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "Warehouse";
    
	def objectType(self):
		return "Warehouse";

File: evidence/generated-systems/rubyonrails/inventory-on-rails/inventoryonrails/app/models/warehouse.rb
class Warehouse < ApplicationRecord


  composed_of :address,
    class_name: "Address",
    mapping: [
      ${$mapping}, 
      ${$mapping}, 
      ${$mapping}, 
      ${$mapping}, 
      %w[address_country country]
    ]

  has_many :StorageLocations, class_name: 'StorageLocation'
  has_many :InventoryItems, class_name: 'InventoryItem'
  has_many :InboundShipments, class_name: 'InboundShipment'
  has_many :OutboundAllocations, class_name: 'OutboundAllocation'
  has_many :OriginTransfers, class_name: 'TransferOrder'
  has_many :DestinationTransfers, class_name: 'TransferOrder'
  has_many :CycleCounts, class_name: 'CycleCount'

end
